lnwallet/btcwallet/btcwallet.go

```diff
@@ -1581,35 +1582,60 @@ func (b *BtcWallet) ListTransactionDetails(startHeight, endHeight int32,
 			currentHeight, blockPackage, b.netParams,
 		)
 		if err != nil {
-			return nil, err
+			return nil, 0, 0, err
 		}
 
 		txDetails = append(txDetails, details...)
 	}
 	for _, tx := range txns.UnminedTransactions {
 		detail, err := unminedTransactionsToDetail(tx, b.netParams)
 		if err != nil {
-			return nil, err
+			return nil, 0, 0, err
 		}
 
 		txDetails = append(txDetails, detail)
 	}
 
+	var (
+		firstIdx = indexOffset + 1
+		lastIdx  = int(indexOffset) + maxTransactions
+	)
+
 	// Return empty transaction list, if offset is more than all
 	// transactions.
 	if int(indexOffset) >= len(txDetails) {
-		return []*lnwallet.TransactionDetail{}, nil
+		firstIdx = 0
+		lastIdx = 0
+		txDetails = []*lnwallet.TransactionDetail{}
+
+		return txDetails, uint64(firstIdx), uint64(lastIdx), nil
 	}
 
-	end := int(indexOffset) + maxTransactons
+	// If maxTransactions is set to -1, then we'll return all transactions
+	// starting from the offset.
+	if maxTransactions == -1 {
+		lastIdx = len(txDetails)
```

Comment:
This is incorrect. The last index is `len(txDetails)-1`. But for the slice access (`[a:b]`) it's correct, since `b` is exclusive.

Reply:
Unfortunately the example [from the invoices](https://github.com/lightningnetwork/lnd/blob/5a8026aba9a4601572e4269281564d662d905fc5/channeldb/invoices.go#L604) is a bad one. There the add index has a special meaning and zero is kind of a special case...

But an index in a pagination should always return the exact indexes of the first and last entry. Then my pagination query can just look something like this (pseudo code):

```
offset := 0
pageSize := 20
result := nil

for {
  result = query(offset, pageSize)
  if len(result) < pageSize {
    return
  }

  offset = result.last_index + 1
}

```